feat: implement http basic authentication for redfish api's

DevipriyaS17 opened the pull request:
1. Added basic auth script to update OpenAPI spec and regenerate authentication-related code.
2. Makefile updated with rf-auth target to execute the authentication script.
3. Implemented HTTP Basic Auth middleware with credential validation.
4. Added UT for  BasicAuthValidator  -  100.0% , AuthMiddleware - 100.0%

nmgaston committed:
chore: add sync.yml for auto sync

nmgaston committed:
Merge branch 'main' into addSyncToDMTRepo

DevipriyaS17 committed:
chore: update sync.yml

DevipriyaS17 committed:
Merge pull request #2 from nmgaston/addSyncToDMTRepo

chore: add sync.yml for auto sync

nmgaston committed:
Merge branch 'device-management-toolkit:main' into main

sudhir-intc committed:
feat: implement redfish plugin architecture with openapi generation

- Add plugin architecture system for DMT extensibility
- Implement Redfish v1 API with DMTF specifications
- Add OpenAPI code generation build system with makefile targets
- Include reference Redfish endpoint implementations:
  - Service Root, Systems, Chassis, Managers collections
  - Computer System reset actions and power state management
- Integrate with DMT shared infrastructure (config, logging, database)
- Include DMTF Redfish schema files for v2025.3 compliance

nmgaston committed:
feat: implement Redfish plugin with Base.1.22.0 message registry

- Add Redfish plugin architecture using OpenAPI-generated handlers
- Implement DMTF Redfish v1 ComputerSystem endpoint
- Add Base.1.22.0 message registry for error responses
- Integrate registry lookup system for Redfish-compliant errors
- Add power state management with conflict detection
- Support reset types: On, ForceOff, ForceRestart, PowerCycle
- Add WSMAN backend integration for AMT devices

nmgaston committed:
docs: update README and logger configuration for Redfish plugin

nmgaston committed:
Merge branch 'redfish-plugin-arch' into mapRegistryFiles

nmgaston committed:
refactor: implement Base.1.22.0 registry lookup and remove error fallbacks

nmgaston committed:
Merge branch 'mapRegistryFiles' of https://github.com/nmgaston/console into mapRegistryFiles

nmgaston committed:
fix(redfish): return correct HTTP status codes for invalid requests

nmgaston committed:
refactor(handler): consolidate error handling with lookup table pattern

nmgaston committed:
Merge branch 'device-management-toolkit:main' into main

nmgaston committed:
Merge branch 'main' into mapRegistryFiles

sudhir-intc committed:
Merge remote-tracking branch 'upstream/main' into redfish-plugin-arch

sudhir-intc committed:
Merge branch 'main' into redfish-plugin-arch

nmgaston committed:
Merge branch 'redfish-plugin-arch' into mapRegistryFiles

sudhir-intc committed:
chore: merge pull request #9 from nmgaston/mapRegistryFiles

feat: map registry files for Base 1.22.0 error messages

sudhir-intc committed:
revert: revert files not changed in this PR

DevipriyaS17 committed:
feat: HTTP basic authentication code changes

DevipriyaS17 committed:
Merge branch 'redfish-plugin-arch' into basic_Auth

sudhir-intc committed:
fix: replace stub implementation of Systems and SystemsId

DevipriyaS17 committed:
feat: fix lint errors

DevipriyaS17 committed:
feat: fix format and lint issues

DevipriyaS17 committed:
feat: remove the duplication of code

sudhir-intc commented on redfish/plugin.go:
```diff
@@ -90,24 +102,63 @@ func (p *Plugin) RegisterMiddleware(ctx *plugin.Context) error {
+		redfishgenerated.RegisterHandlersWithOptions(ctx.Router, p.server, redfishgenerated.GinServerOptions{
+			BaseURL:      "",
+			ErrorHandler: p.createErrorHandler(),
+			Middlewares: []redfishgenerated.MiddlewareFunc{
```
The current authentication middleware logic (lines 120–140) could be improved for better readability and unit testing. Consider extracting this logic into a named function, e.g., RedfishAuthMiddleware, and managing public endpoints via a set or map for easier updates and testability. This separation will make the code more maintainable and easier to unit test

sudhir-intc replied:
Example Refactor
```go
// RedfishAuthMiddleware returns a middleware that applies authentication rules as per OpenAPI spec.
func RedfishAuthMiddleware(basicAuth gin.HandlerFunc, publicEndpoints map[string]bool) gin.HandlerFunc {
    return func(c *gin.Context) {
        path := c.Request.URL.Path

        if publicEndpoints[path] {
            c.Next()
            return
        }

        // Protect all /redfish/v1/* endpoints except explicitly public
        if strings.HasPrefix(path, "/redfish/v1/") {
            basicAuth(c)
            return
        }

        // Default: no authentication
        c.Next()
    }
}

// In RegisterRoutes:
publicEndpoints := map[string]bool{
    "/redfish/v1/":         true,
    "/redfish/v1/$metadata": true,
}

redfishgenerated.RegisterHandlersWithOptions(ctx.Router, p.server, redfishgenerated.GinServerOptions{
    BaseURL:      "",
    ErrorHandler: p.createErrorHandler(),
    Middlewares: []redfishgenerated.MiddlewareFunc{
        RedfishAuthMiddleware(basicAuthMiddleware, publicEndpoints),
    },
})

```

DevipriyaS17 replied:
Updated the code, thank you

sudhir-intc reviewed (commented):
Overall LGTM with a improvement as suggested.
Please also add the UT code.

sudhir-intc committed:
refactor: updated usecase interfaces with ctx and implemented /redfish/v1/Systems and /redfish/v1/Systems/{Systemid}

sudhir-intc committed:
refactor: updated usecase interfaces with ctx and wsman repo impl

1. implemented /redfish/v1/Systems & 2. /redfish/v1/Systems/{Systemid}
2. added ctx for all the use case interfaces so that the logger can be used

sudhir-intc committed:
refactor: improved the wsman_repo code w.r.t readability and helper funcs

sudhir-intc committed:
Merge branch 'redfish-plugin-arch' of https://github.com/nmgaston/console into redfish-plugin-arch

sudhir-intc committed:
fix: lint warnings and errors fixed

sudhir-intc committed:
Merge branch 'redfish-plugin-arch' of https://github.com/nmgaston/console into redfish-plugin-arch

amarnath-ac committed:
refactor: remove manager and chassis endpoints from Redfish API

- Removed manager and chassis endpoints from OpenAPI specification
- Updated generated server code and types accordingly

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

amarnath-ac committed:
refactor: remove chassis and manager spec files

Remove chassis and manager related OpenAPI specifications and update route handlers to exclude these endpoints from the service root response.

sudhir-intc committed:
Merge branch 'redfish-plugin-arch' into remove_spec_manager_chassis

sudhir-intc committed:
chore: merge pull request #13 from nmgaston/remove_spec_manager_chassis

Remove manager and chassis endpoints from Redfish API

sudhir-intc committed:
fix: merge issues for build and lint

sudhir-intc commented on redfish/internal/controller/http/v1/handler/middleware.go:
We need to add this middleware as part of the Redfish Plugin as this is specific to Redfish

DevipriyaS17 replied:
moved as part of plugin

DevipriyaS17 committed:
Merge branch 'redfish-plugin-arch' into basic_Auth

DevipriyaS17 committed:
feat: addressed review comments and updated unit test

DevipriyaS17 committed:
feat: fix lint issues

sudhir-intc committed:
refactor: ensured in the openapi spec the component names are sorted alphabetically

sudhir-intc committed:
Merge branch 'redfish-plugin-arch' into basic_Auth

sudhir-intc committed:
fix: corrected the generated redfish-openapi.yaml file

sudhir-intc committed:
Merge branch 'redfish-plugin-arch' into basic_Auth
